### PR TITLE
Deny rust_2018_idioms globally

### PR DESCRIPTION
**Pull request by phansch**
cc https://github.com/rust-lang/rust/issues/58099#issuecomment-484921194

### CODE REVIEW

**Comment by phansch** [comment]
Looks like `src/liballoc/tests/` is still missing `#![deny(rust_2018_idioms)]`. I'll see if I can fix it.

**Comment by phansch** [comment]
I'm not sure if there are alternatives to 8e55559. Apart from that, I think this PR is ready.

File: src/liballoc/tests/lib.rs [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #![feature(try_reserve)]
 #![feature(unboxed_closures)]
 #![feature(vecdeque_rotate)]
+#![deny(rust_2018_idioms)]
```
Comment (petrochenkov) [inline src/liballoc/tests/lib.rs]: I don't understand why it's denied here if it was denied in rustbuild already.
Also, why aren't all the `deny(rust_2018_idioms)`s in specific crates removed?

Comment (Centril) [reply src/liballoc/tests/lib.rs]: We can remove them in a follow up. However, I personally prefer to keep them as it works better when you `cargo check` a crate without `x.py`.

File: src/libcore/tests/lib.rs [hunk 2]
```diff
@@ -32,8 +32,8 @@
 #![feature(slice_partition_dedup)]
 #![feature(copy_within)]
 #![feature(int_error_matching)]
+#![deny(rust_2018_idioms)]
 
-extern crate core;
 extern crate test;
```
Comment (RalfJung) [inline src/libcore/tests/lib.rs]: Any idea why you didn't also have to remove this line? It causes a build failure when building the tests "from the outside" at https://travis-ci.org/RalfJung/miri-test-libstd/builds/523334973.

Comment (phansch) [reply src/libcore/tests/lib.rs]: I'm not really sure. Do you maybe have to pass `--test` in order for the crate to be used?

Comment (RalfJung) [reply src/libcore/tests/lib.rs]: Hm, no... removing it actually makes it fail in `x.py`.  But somehow it is considered redundant when compiled in a less magic environment?

Comment (phansch) [reply src/libcore/tests/lib.rs]: I think adding `#[allow(unused_extern_crates)]` before `extern crate test;` should fix it (like [here](https://github.com/rust-lang/rust/blob/d6f513ec7d1e83c8689f94fb48686dd11f1d1c80/src/librustc/lib.rs#L94)), but I won't be able to actually try it out until later today.

Comment (RalfJung) [reply src/libcore/tests/lib.rs]: This should also work: https://github.com/rust-lang/rust/pull/60201

**Pull request merged**